priority/linux.go

```diff
@@ -34,32 +34,31 @@ const (
 	IOPrioWhoUser
 )
 
+const selfPID = 0
+
 // SetNice sets the unix "nice" value of the current process
 func SetNice(priority int) error {
 	var err error
-	// pid 0 means "self"
-	pid := 0
 
 	if priority < -20 || priority > 19 {
 		return fmt.Errorf("unexpected priority value %d", priority)
 	}
 
-	currentPriority, _ := unix.Getpriority(unix.PRIO_PROCESS, 0)
-	if currentPriority == priority {
-		return nil
-	}
-
 	// Move ourselves to a new process group so that we can use the process
-	// group variants of Setpriority etc to affect all of our threads in one go
-	err = unix.Setpgid(pid, 0)
+	// group variants of Setpriority to affect all of our processes at once
+	err = unix.Setpgid(selfPID, 0)
 	if err != nil {
-		return fmt.Errorf("cannot set process group priority, restic will run with the default priority: %w", err)
+		return fmt.Errorf("cannot set process group, restic will run with the default priority: %w", err)
 	}
 
-	clog.Debugf("setting process priority to %d", priority)
-	err = unix.Setpriority(unix.PRIO_PROCESS, pid, priority)
+	clog.Debugf("setting group process priority to %d", priority)
+	err = unix.Setpriority(unix.PRIO_PGRP, selfPID, priority)
 	if err != nil {
-		return fmt.Errorf("cannot set process priority, restic will run with the default priority: %w", err)
+		clog.Debugf("setting process priority to %d instead", priority)
+		err = unix.Setpriority(unix.PRIO_PROCESS, selfPID, priority)
+		if err != nil {
+			return fmt.Errorf("cannot set process priority, restic will run with the default priority: %w", err)
+		}
```

Comment:
I know that your intent is

> Try to be more resilient in case of a failure

but does this really help? As a user, I'm setting the niceness in order to not have the restic run impact other processes on the system too much. But the thing that's using the resources is restic, not resticprofile. Changing just resticprofile's priority doesn't solve the problem, in my opinion.

If you ask me, resticprofile should not consider "well, at least I could set _my own_ priority" a success and instead flat out stop running if my desired configuration could not be applied. Because otherwise, people won't notice. I mean, as far as I can tell, the broken implementation has been in resticprofile for four years and people didn't notice ;)

(Also, you're no longer logging the original `err` that prevented the process group `Setpriority` call.)

Reply:
Hm, makes sense. But doesn't https://github.com/creativeprojects/resticprofile/issues/229#issuecomment-1859349273 (i.e. the report of restic _not_ running with the configured niceness) show that the priority is not being inherited, at least in some cases?

Reply:
That's right, but I don't really understand why 🤔

Reply:
Me neither, sorry :man_shrugging:

I suppose the code as you're proposing it fixes the `PRIO_PGRP` issue and the fallback doesn't make things _worse_ compared to before, so … good enough for me :wink: